### PR TITLE
Update to the new distro mechanism

### PR DESCRIPTION
**Pull request by blueraft**

### CODE REVIEW

File: template_README.md [hunk 1]
```diff
@@ -2,24 +2,10 @@
...
-> token (PAT) according to the instructions in the GitHub docs [here](https://docs.github.com/en/packages/working-with-a-github-packages-registry/working-with-the-container-registry#authenticating-with-a-personal-access-token-classic).
-> If you want to make the image public (recommended), you should make sure that your 
-> organization settings allow public packages and make this package public after building it.
-> You can read more about this in the GitHub docs [here](https://docs.github.com/en/packages/learn-github-packages/configuring-a-packages-access-control-and-visibility).
-
-> [!TIP]
-> In order for others to find and learn from your distribution we in FAIRmat would
-> greatly appreciate it if you would add the topic `nomad-distribution` by clicking the
-> ⚙️ next to "About" on the main GitHub page for this repository.
-
```
Comment (hampusnasstrom) [inline template_README.md:22]: What was the reason for removing this? I think it can be quite useful information as you are setting up the distribution. It's easy enough for the user to delete afterwards.

File: requirements.txt
Comment (hampusnasstrom) [inline requirements.txt:1]: Should we really include a `requirements.txt` with all the optional dependencies as default?

File: pyproject.toml [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+[project]
+classifiers = [
+  "Intended Audience :: Developers",
+  "Operating System :: OS Independent",
+  "Programming Language :: Python",
+  "Programming Language :: Python :: 3.9",
+  "Programming Language :: Python :: 3.10",
+  "Programming Language :: Python :: 3.11",
+]
+name = "nomad-distribution"
```
Comment (hampusnasstrom) [inline pyproject.toml:14]: Should we attempt to modify this value when we generate the template instance?

File: nomad-oasis.zip
Comment (hampusnasstrom) [inline nomad-oasis.zip:1]: Why did you remove this file? Are you generating it with the Github workflow instead?

File: configs/nomad.yaml [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+    materials_index: nomad_oasis_materials_v1
+
+plugins:
+  options:
+    schemas/nomad_material_processing:
+      python_package: nomad_material_processing
+    schemas/nomad_measurements:
+      python_package: nomad_measurements
+    parsers/nomad_measurements/xrd:
+      python_package: nomad_measurements.xrd.parser
```
Comment (hampusnasstrom) [inline configs/nomad.yaml:34]: This should be deleted with the new plugin mechanism.
```suggestion
```

File: template_README.md [hunk 4]
```diff
@@ -28,43 +14,46 @@ Below are instructions for how to deploy this NOMAD Oasis distribution
...
+   Docker nowadays comes with `docker compose` built in. Prior, you needed to
+   install the stand-alone [docker-compose](https://docs.docker.com/compose/install/).
+2. _On Linux only,_ recursively change the owner of the `.volumes` directory to the nomad user (1000)
+
 ```sh
 sudo chown -R 1000 .volumes
 ```
-5. Pull the images specified in the `docker-compose.yaml` (retrieved from the `nomad-oasis.zip`).
+
+5. Pull the images specified in the `docker-compose.yaml`
```
Comment (hampusnasstrom) [inline template_README.md:25]: Step 3-4? I think we should still have a compressed archive that the users can just curl.

**Review by hampusnasstrom — CHANGES_REQUESTED** [review]
I'm not sure about the changes to the README. Peraps we discuss this over a call.

File: template_README.md (hunk 4 above)
Comment (blueraft) [reply template_README.md:39]: The docker compose file and the configs are part of the repo now, why'd an curl step be necessary?

File: pyproject.toml (hunk 2 above)
Comment (blueraft) [reply pyproject.toml:14]: We can but this value doesn't really make it to the app.

File: nomad-oasis.zip
Comment (blueraft) [reply nomad-oasis.zip:1]: I mentioned in the other commit, but the values in the files are part of the repo so there'd be no need to download and unzip the archive.

File: requirements.txt
Comment (blueraft) [reply requirements.txt:1]: Yes, all of the requirements in the file are the direct plugin dependencies and the transitive dependencies. 

Since this is an app, we want the builds to be deterministic and reproducible even six months or two years from now. If they're not locked, the latest versions will change over time, etc.

Some discussion on this here: https://doc.rust-lang.org/cargo/faq.html#why-have-cargolock-in-version-control

File: nomad-oasis.zip
Comment (hampusnasstrom) [reply nomad-oasis.zip:1]: But you wouldn't clone this repo on your production server, or?

**Review by hampusnasstrom — APPROVED** [review]
Great work!

File: template_README.md [hunk 5]
```diff
@@ -28,43 +28,53 @@ Below are instructions for how to deploy this NOMAD Oasis distribution
...
+
+2. Clone the repository or download the repository as a zip file.
+
 ```sh
-unzip nomad-oasis.zip
-cd nomad-oasis
+git clone https://github.com/GITHUB_REPOSITORY.git
 ```
-4. _On Linux only,_ recursively change the owner of the `.volumes` directory to the nomad user (1000) 
+
```
Comment (blueraft) [inline template_README.md:39]:
```suggestion
or
```sh
curl-L -o nomad-oasis.zip "https://github.com/GITHUB_REPOSITORY/archive/main.zip"

unzip nomad-oasis.zip
cd nomad-oasis
```

@hampusnasstrom what do you think?
